From 1e7ea5487203988c0304d2788b35d002bc5e89f9 Mon Sep 17 00:00:00 2001
From: "G. Branden Robinson" <g.branden.robinson@gmail.com>
Date: Thu, 9 Apr 2026 00:49:47 -0500
Subject: [libgroff]: Drop "matherr.c" file.

...declaring (under the preprocessor conditions `HAVE_STRUCT_EXCEPTION`
and `TLOSS`) `matherr()` and `struct exception`.  The former is K&R-
style, thus prohibited in ISO C23, and can (should) cause build failures
with compilers configured for that version of the C language.  Autoconf
2.73, released last month, tries harder to arrange builds to use C23 by
default.  Also we straight up don't _need_ this declaration (anymore);
nothing in groff calls `matherr()`.  It dates back to the Clark days,
when there was no gnulib for groff to use (or maybe it didn't support
C++ projects at the time).  Whatever gnulib needs, it transitively
closures for us, so we no longer need to and can throw this 30-year old
stuff of ours away.

* src/libs/libgroff/matherr.c: Delete.

* src/libs/libgroff/libgroff.am (libgroff_a_SOURCES): Drop
  aforementioned file from macro definition.

Fixes <https://savannah.gnu.org/bugs/?68220>.  Thanks to Alexis
Hildebrandt for the report.  Also continues the long process of fixing
Savannah #66518.

src/libs/libgroff/.gitignore: Stop ignoring its compiled object file.
---
 ChangeLog                     | 24 +++++++++++++++++++
 src/libs/libgroff/.gitignore  |  1 -
 src/libs/libgroff/libgroff.am |  2 --
 src/libs/libgroff/matherr.c   | 55 -------------------------------------------
 4 files changed, 24 insertions(+), 58 deletions(-)
 delete mode 100644 src/libs/libgroff/matherr.c

diff --git a/src/libs/libgroff/libgroff.am b/src/libs/libgroff/libgroff.am
index 25b705d2a..b49958096 100644
--- a/src/libs/libgroff/libgroff.am
+++ b/src/libs/libgroff/libgroff.am
@@ -47,7 +47,6 @@ libgroff_a_SOURCES = \
   src/libs/libgroff/lineno.cpp \
   src/libs/libgroff/localcharset.c \
   src/libs/libgroff/macropath.cpp \
-  src/libs/libgroff/matherr.c \
   src/libs/libgroff/maxfilename.cpp \
   src/libs/libgroff/maxpathname.cpp \
   src/libs/libgroff/mksdir.cpp \
@@ -165,7 +164,6 @@ uninstall_charset_data:
 	  $(RM) $(DESTDIR)$(libdir)/t-charset.alias; \
 	fi
 
-
 # Local Variables:
 # mode: makefile-automake
 # fill-column: 72
diff --git a/src/libs/libgroff/matherr.c b/src/libs/libgroff/matherr.c
deleted file mode 100644
index b1ca4fcb6..000000000
--- a/src/libs/libgroff/matherr.c
+++ /dev/null
@@ -1,54 +0,0 @@
-/* Copyright 1989-2020 Free Software Foundation, Inc.
-     Written by James Clark (jjc@jclark.com)
-
-This file is part of groff, the GNU roff typesetting system.
-
-groff is free software; you can redistribute it and/or modify it under
-the terms of the GNU General Public License as published by the Free
-Software Foundation, either version 3 of the License, or
-(at your option) any later version.
-
-groff is distributed in the hope that it will be useful, but WITHOUT ANY
-WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
-for more details.
-
-You should have received a copy of the GNU General Public License
-along with this program.  If not, see <http://www.gnu.org/licenses/>. */
-
-#ifdef HAVE_CONFIG_H
-#include <config.h>
-#endif
-
-#include <math.h>
-#include <errno.h>
-
-#ifdef HAVE_STRUCT_EXCEPTION
-#ifdef TLOSS
-
-int matherr(exc)
-struct exception *exc;
-{
-  switch (exc->type) {
-  case SING:
-  case DOMAIN:
-    errno = EDOM;
-    break;
-  case OVERFLOW:
-  case UNDERFLOW:
-  case TLOSS:
-  case PLOSS:
-    errno = ERANGE;
-    break;
-  }
-  return 1;
-}
-
-#endif /* TLOSS */
-#endif /* HAVE_STRUCT_EXCEPTION */
-
-// Local Variables:
-// fill-column: 72
-// mode: C++
-// End:
-// vim: set cindent noexpandtab shiftwidth=2 textwidth=72:
cgit v1.3

From 4e6c98f3df065743e5598c158d71af7fc8f63a25 Mon Sep 17 00:00:00 2001
From: "G. Branden Robinson" <g.branden.robinson@gmail.com>
Date: Thu, 9 Apr 2026 00:59:58 -0500
Subject: [build]: Drop obsolescent configuration code.

Drop no longer needed check for `struct exception`.  It was used only by
the now-deleted "src/libs/libgroff/matherr.c".

* m4/groff.m4 (GROFF_STRUCT_EXCEPTION): Delete.

* configure.ac: Drop interpolation of `GROFF_STRUCT_EXCEPTION`.

* Makefile.am: Stop advertising availability of `HAVE_STRUCT_EXCEPTION`
  preprocessor symbol.
---
 ChangeLog    | 10 ++++++++++
 Makefile.am  |  1 -
 configure.ac |  1 -
 m4/groff.m4  | 19 -------------------
 4 files changed, 10 insertions(+), 21 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 7ae89f651..4d475d4e3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -327,7 +327,6 @@ man7dir=$(manroot)/man$(man7ext)
 # HAVE_DECL_GETC_UNLOCKED	if you have getc_unlocked()
 # HAVE_DECL_SYS_SIGLIST		if you have sys_siglist[]
 #
-# HAVE_STRUCT_EXCEPTION		if <math.h> defines struct exception
 # HAVE_SYS_NERR			if you have sysnerr in <errno.h> or <stdio.h>
 # HAVE_SYS_ERRLIST		if you have sys_errlist in <errno.h> or
 #				<stdio.h>
diff --git a/configure.ac b/configure.ac
index 46e1cdd94..db113f81c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -104,7 +104,6 @@ GROFF_UNSIGNED_LONG_LONG
 GROFF_UINTMAX_T
 GROFF_TIME_T
 GROFF_TYPE_SIGNAL
-GROFF_STRUCT_EXCEPTION
 
 # check endianness
 AC_C_BIGENDIAN
diff --git a/m4/groff.m4 b/m4/groff.m4
index fbe3d07f3..e502524c2 100644
--- a/m4/groff.m4
+++ b/m4/groff.m4
@@ -802,25 +802,6 @@ struct tm *p = localtime(&t);
 	[Define if localtime() takes a long * not a time_t *.])])
    AC_LANG_POP([C++])])
 
-AC_DEFUN([GROFF_STRUCT_EXCEPTION],
-  [AC_MSG_CHECKING([struct exception])
-   AC_COMPILE_IFELSE([
-       AC_LANG_PROGRAM([[
-
-#include <math.h>
-
-       ]],
-       [[
-
-struct exception e;
-
-       ]])
-     ],
-     [AC_MSG_RESULT([yes])
-      AC_DEFINE([HAVE_STRUCT_EXCEPTION], [1],
-	[Define if <math.h> defines struct exception.])],
-     [AC_MSG_RESULT([no])])])
-
 AC_DEFUN([GROFF_ARRAY_DELETE],
   [AC_LANG_PUSH([C++])
    AC_MSG_CHECKING([whether ISO C++98 array deletion syntax is supported])
-- 
cgit v1.3

